connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -24,24 +24,32 @@ import scala.collection.JavaConverters._
  */
 class UCSingleCatalog extends TableCatalog with SupportsNamespaces {
 
-  private var deltaCatalog: TableCatalog = null
+  @volatile private var delegate: TableCatalog = null
 
   override def initialize(name: String, options: CaseInsensitiveStringMap): Unit = {
     val proxy = new UCProxy()
     proxy.initialize(name, options)
-    deltaCatalog = Class.forName("org.apache.spark.sql.delta.catalog.DeltaCatalog")
-      .getDeclaredConstructor().newInstance().asInstanceOf[TableCatalog]
-    deltaCatalog.asInstanceOf[DelegatingCatalogExtension].setDelegateCatalog(proxy)
+    try {
+      if (options.containsKey("__TEST_NO_DELTA__")) {
+        throw new ClassNotFoundException("DeltaCatalog")
+      }
+      delegate = Class.forName("org.apache.spark.sql.delta.catalog.DeltaCatalog")
+        .getDeclaredConstructor().newInstance().asInstanceOf[TableCatalog]
+      delegate.asInstanceOf[DelegatingCatalogExtension].setDelegateCatalog(proxy)
+    } catch {
+      case _: ClassNotFoundException =>
```

Comment:
can you add logs so that we know what is happening? otherwise this will be impossible to debug in real deployment.